src/components/TextField/TextField.jsx

```diff
@@ -0,0 +1,75 @@
+import { TextField as MUITextField } from '@mui/material';
+import InputAdornment from '@mui/material/InputAdornment';
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+import textFieldStyles from './TextFieldStyles';
+
+const TextField = (props) => {
+    const {
+        placeholder,
+        error,
+        position,
+        children,
+        adornment,
+        disabled,
+        helperText,
+        label,
+        type,
+        className,
+        ...rest
+    } = props;
+
+    const [isFocused, setIsFocused] = useState(false);
+
+    const onFocusHandler = () => {
```

Comment:
ur component shouldnt have any handlers in itself, you should apply handler that comes as prop